### PR TITLE
DOC-9465for7.0: Added security checklist

### PR DESCRIPTION
**Pull request by tonyjhillman**

### CODE REVIEW

**Comment by tonyjhillman** [comment]
Approved by Ian McCloy as per https://github.com/couchbase/docs-server/pull/2400.

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 1]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+
+[cols="5,15"]
+|===
+|
+Access control
+a|
+* [ ] Create unique user accounts for each individual and application that accesses the platform.
+
+* [ ] Implement Role-Based Access Control and assign roles following a principle of least privilege
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:57]:
```suggestion
* [ ] Implement Role-Based Access Control and assign roles following a principle of least privilege.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 2]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Leverage strongest available authentication mechanisms
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:59]:
```suggestion
* [ ] Leverage strongest available authentication mechanisms.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 3]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Ensure secure storage and transfer of credentials or certificates
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:61]:
```suggestion
* [ ] Ensure secure storage and transfer of credentials or certificates.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 4]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Implement Multi-Factor Authentication for individual access
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:63]:
```suggestion
* [ ] Implement Multi-Factor Authentication for individual access.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 5]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+|
+Secure Network Communication
+a|
+* [ ] Replace self-signed certificates with external CA certificates
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:68]:
```suggestion
* [ ] Replace self-signed certificates with external CA certificates.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 6]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Ensure all applications are using encrypted connections only and block insecure ports
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:70]:
```suggestion
* [ ] Ensure all applications are using encrypted connections only and block insecure ports.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 7]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Enforce TLS with ‘strict’ encryption setting to encrypt all network traffic internally and externally.
+
+* [ ] Only Enable IP Address Families as needed
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:74]:
```suggestion
* [ ] Only Enable IP Address Families as needed.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 8]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+
+|
+Secure Data Storage
+
+a|
+* [ ] Deploy encryption at rest
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:81]:
```suggestion
* [ ] Deploy encryption at rest.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 9]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Implement (FLE) Field-level Encryption
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:83]:
```suggestion
* [ ] Implement (FLE) Field-level Encryption.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 10]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Configure Secrets Management
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:85]:
```suggestion
* [ ] Configure Secrets Management.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 11]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+|
+Limit Data Exposure
+
+a|
+* [ ] Remove unnecessary services and software
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:91]:
```suggestion
* [ ] Remove unnecessary services and software.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 12]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Regularly scan for and protect sensitive data
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:93]:
```suggestion
* [ ] Regularly scan for and protect sensitive data.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 13]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Control geographic distribution of data
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:95]:
```suggestion
* [ ] Control geographic distribution of data.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 14]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Leverage log redaction when collecting and transferring logs
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:97]:
```suggestion
* [ ] Leverage log redaction when collecting and transferring logs.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 15]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Deploy a comprehensive DR strategy utilizing offsite backups
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:99]:
```suggestion
* [ ] Deploy a comprehensive DR strategy utilizing offsite backups.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 16]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+|
+Auditing
+
+a|
+* [ ] Enable and configure auditing
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:105]:
```suggestion
* [ ] Enable and configure auditing.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 17]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Review audit logs manually and programmatically for anomalies
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:107]:
```suggestion
* [ ] Review audit logs manually and programmatically for anomalies.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 18]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+|
+Regular Review
+
+a|
+* [ ] Assess security from core to edge and perform regular security health checks
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:113]:
```suggestion
* [ ] Assess security from core to edge and perform regular security health checks.
```

File: modules/manage/pages/manage-security/security-management-overview.adoc [hunk 19]
```diff
@@ -38,3 +38,83 @@ Information is provided in xref:manage:manage-logging/manage-logging.adoc[Manage
...
+
+* [ ] Review enhancements provided with each new version of the Couchbase Data Platform
```
Comment (RayOffiah) [inline modules/manage/pages/manage-security/security-management-overview.adoc:115]:
```suggestion
* [ ] Review enhancements provided with each new version of the Couchbase Data Platform.
```

**Review by RayOffiah — APPROVED** [review]
I'd probably add full stops after each list item, but aside from that small point, looks good to me.

File: modules/manage/pages/manage-security/security-management-overview.adoc (hunk 1 above)
Comment (tonyjhillman) [reply modules/manage/pages/manage-security/security-management-overview.adoc:57]: I agree. But I've been asked to replicate the checklist exactly, so I've limited myself just to that.

**Pull request merged**